lntest/itest/lnd_amp_test.go

```diff
@@ -15,6 +15,137 @@ import (
 	"github.com/stretchr/testify/require"
 )
 
+// testSendPaymentAMPInvoice tests that we can send an AMP payment to a
+// specified AMP invoice using SendPaymentV2.
+func testSendPaymentAMPInvoice(net *lntest.NetworkHarness, t *harnessTest) {
+	ctxb := context.Background()
+
+	ctx := newMppTestContext(t, net)
+	defer ctx.shutdownNodes()
+
+	const paymentAmt = btcutil.Amount(300000)
+
+	// Set up a network with three different paths Alice <-> Bob. Channel
+	// capacities are set such that the payment can only succeed if (at
+	// least) three paths are used.
+	//
+	//              _ Eve _
+	//             /       \
+	// Alice -- Carol ---- Bob
+	//      \              /
+	//       \__ Dave ____/
+	//
+	ctx.openChannel(ctx.carol, ctx.bob, 135000)
+	ctx.openChannel(ctx.alice, ctx.carol, 235000)
+	ctx.openChannel(ctx.dave, ctx.bob, 135000)
+	ctx.openChannel(ctx.alice, ctx.dave, 135000)
+	ctx.openChannel(ctx.eve, ctx.bob, 135000)
+	ctx.openChannel(ctx.carol, ctx.eve, 135000)
+
+	defer ctx.closeChannels()
+
+	ctx.waitForChannels()
+
+	payAddr := make([]byte, 32)
+	_, err := rand.Read(payAddr)
+	require.NoError(t.t, err)
+
+	addInvoiceResp, err := ctx.bob.AddInvoice(context.Background(), &lnrpc.Invoice{
+		PaymentAddr: payAddr,
+		Value:       int64(paymentAmt),
+		IsAmp:       true,
+	})
+	require.NoError(t.t, err)
+
+	// Increase Dave's fee to make the test deterministic. Otherwise it
+	// would be unpredictable whether pathfinding would go through Charlie
+	// or Dave for the first shard.
+	_, err = ctx.dave.UpdateChannelPolicy(
+		context.Background(),
+		&lnrpc.PolicyUpdateRequest{
+			Scope:         &lnrpc.PolicyUpdateRequest_Global{Global: true},
+			BaseFeeMsat:   500000,
+			FeeRate:       0.001,
+			TimeLockDelta: 40,
+		},
+	)
+	if err != nil {
+		t.Fatalf("dave policy update: %v", err)
+	}
+
+	ctxt, _ := context.WithTimeout(context.Background(), 4*defaultTimeout)
+	payment := sendAndAssertSuccess(
+		ctxt, t, net.Alice,
+		&routerrpc.SendPaymentRequest{
+			PaymentRequest: addInvoiceResp.PaymentRequest,
+			MaxParts:       10,
```

Comment:
if we don't set this, than we will test that we split into parts even by default

Reply:
are you suggesting we remove it from this test as well as the other amp sendpayment itest?

Reply:
Yes. My idea was that after this commit: https://github.com/lightningnetwork/lnd/pull/5253/commits/46671503fd684e74960d3f6e6b50270cd2495c7e we don't have to set this anymore.

Reply:
Done